### PR TITLE
Classifying ADNI Brain Data for Alzheimer's Using Vision Transformer

### PR DESCRIPTION
**Pull request by bquek00**
Alzheimer's disease is a progressive brain disorder that impairs memory and cognitive functions. This project uses a ViT to categorise MRI brain scans into Normal Cognitive (NC) or Alzheimer's disease (AD). I used ViT structure for image processing. First segmenting the input images into 16x16 patches and converts them into vector embeddings. These vectors are then processed using a transformer encoder, followed by a feed-forward neural network for each individual embedding.

### CODE REVIEW

**Comment by LinfengLiu98** [comment]
**This is an initial inspection, no action is required at this point**

Difficulty: Hard

Readme:

- Need more explain on how VIT works and how you developed your algorithm.
- No plots of training and val accuracy
- 

Commit messages: clear, detailed.

Code:
- Uses Vision transformer
- Commented
- Good model design

Functionality/Performance:
- Test accuracy is 73.29%
- Did patient-level split
- Preprocssing data correctly
- No train, val and test split. **Selected model only based on test set results.**


General comments:
Only had one training loss plot to show model performance. Need accuracy plots on both training and val set.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation
Good spacing and comments
Header blocks minimal -1
## Recognition Problem 
Solves problem 
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage, no plots -2
Module present
Commenting
No Data leakage, data splits irregular -1
Difficulty: Hard
## Commit Log
Meaningful commit messages sometimes -1
Progressive commits used
## Documentation
ReadMe acceptable but missing some info -1
Model/technical explanation
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
No Feedback required
Request Description minimal -1
